nautobot/utilities/views.py

```diff
@@ -151,3 +186,1062 @@ def get_return_url(self, request, obj=None):
 
         # If all else fails, return home. Ideally this should never happen.
         return reverse("home")
+
+
+class NautobotViewSetMixin:
+
+    routes = []
+    model = None
+    queryset = None
+    table = None
+    template_name = None
+    filterset = None
+    filterset_form = None
+    import_form = None
+    prefetch_related = []
+
+    def __init__(self):
+        self.routes = self.define_routes()
+
+    @property
+    def detail_queryset(self):
+        return self.model.objects.all()
+
+    @property
+    def table_queryset(self):
+        return self.model.objects.prefetch_related(*self.prefetch_related)
+
+    def define_routes(self):
+        return []
+
+    def get_extra_context(self, request, view_type, instance=None):
+        """
+        Return any additional context data for the template.
+        request: The current request
+        instance: The object being viewed
+        """
+        return {}
+
+    def get_template_name(self, view_type):
+        # Use "<app>/<model>_<view_type> if available, else fall back to generic templates
+        model_opts = self.model._meta
+        app_label = model_opts.app_label
+        if view_type == "detail":
+            return f"{app_label}/{model_opts.model_name}.html"
+
+        try:
+            select_template([f"{app_label}/{model_opts.model_name}_{view_type}.html"])
+            return f"{app_label}/{model_opts.model_name}_{view_type}.html"
```

Comment:
Yeah if it is just a matter of adding `_detail.html` to every detail template name I think it def doable. But I can see it going either way.

Reply:
Since this is aligning to a more convention-esque approach, I'm in favor of encouraging a bit further of clarity and requiring `_detail` in the template name.

If you're looking to migrate to using NautobotViewSet, might as well go the extra step.

It's not beneficial to permit it now, only to get more strict in the future.